charts/pihole/templates/podmonitor.yaml

```diff
@@ -7,8 +7,8 @@ metadata:
     chart: {{ template "pihole.chart" . }}
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
-    {{- if .Values.monitoring.podMonitor.labels }}
-    {{ toYaml .Values.monitoring.podMonitor.labels }}
+    {{- with .Values.monitoring.podMonitor.labels }}
+    {{- . | toYaml | nindent 4 }}
```

Comment:
Note that:
- `with` behaves like an `if` statement as well alongside updating the context
- `nindent` will add a new line before indenting whatever is piped into it - it helps us retain some indentation in our Helm template file

Reply:
Nice, I learned something new today